File: src/features/camera/types/scene.ts
import type { Coordinate } from '@/features/map/types/coordinate';
import type { ImageClass, TripPhoto } from '@/features/trip-session/types/trip';

/**
 * Para que a foto foi tirada; decide se ela é guardada (RF-22).
 *
 * `context`: leitura automática, só vira classe para o Random Forest.
 * `tourist_spot`: "Registrar ponto turístico"; essa foto fica guardada.
 */
export type ScenePurpose = 'context' | 'tourist_spot';

/** O formato de `POST /v1/trips/{id}/scenes`. */
export type SceneResult = {
  imageClass: ImageClass;
  confidence: number;
  probabilities: Record<string, number>;
  /** `false` quando a leitura não virou evento (cena repetida ou incerta). */
  recorded: boolean;
  reason: string | null;
  eventId: string | null;
  photo: TripPhoto | null;
};

export type SendSceneParams = {
  tripId: string;
  /** Arquivo local devolvido pela câmera. */
  uri: string;
  purpose: ScenePurpose;
  location?: Coordinate | null;
  signal?: AbortSignal;
};

File: backend/app/schemas/scene.py
"""Classificação de cena (RF-16, RF-22): o que o app envia e o que recebe."""

from enum import StrEnum
from uuid import UUID

from app.schemas.base import ApiModel
from app.schemas.trip import ImageClass, Photo


class ScenePurpose(StrEnum):
    """Para que a foto foi tirada. Decide se ela é guardada ou não."""

    # Leitura automática da câmera: só a classe fica, a foto é descartada.
    CONTEXT = "context"
    # "Registrar ponto turístico": a foto é guardada.
    TOURIST_SPOT = "tourist_spot"


class SceneResponse(ApiModel):
    image_class: ImageClass
    confidence: float
    # Probabilidade de cada uma das 4 classes.
    probabilities: dict[str, float]
    # `false` quando a leitura não virou evento (cena repetida ou confiança baixa).
    recorded: bool
    # `low_confidence` ou `unchanged`, quando não foi registrada.
    reason: str | None = None
    event_id: UUID | None = None
    # Só no "registrar ponto turístico".
    photo: Photo | None = None
